src/main/java/com/odysseusinc/arachne/datanode/service/messaging/PathwayRequestHandler.java

```diff
@@ -0,0 +1,179 @@
+    private final Template pathwaysRunnerTemplate;
+    private static final int PAGE_SIZE = 10000;
+    public static final String PATHWAY_BUILD_ERROR = "Failed to build Pathway data";
+    public static final Logger LOGGER = LoggerFactory.getLogger(PathwayRequestHandler.class);
+
+    @Autowired
+    public PathwayRequestHandler(SqlRenderService sqlRenderService,
+																 AtlasService atlasService,
+																 CommonEntityService commonEntityService,
+																 GenericConversionService conversionService,
+																 CentralSystemClient centralClient,
+																 Template pathwaysRunnerTemplate) {
+
+        super(sqlRenderService, atlasService);
+        this.commonEntityService = commonEntityService;
+        this.conversionService = conversionService;
+        this.centralClient = centralClient;
+			this.pathwaysRunnerTemplate = pathwaysRunnerTemplate;
+		}
+
+    @Override
+    public List<CommonPathwayDTO> getObjectsList(List<Atlas> atlasList) {
+
+    		List<Atlas> atlases27 = AtlasUtils.filterAtlasByVersion27(atlasList);
+        List<Pathway> pathways = atlasService.<AtlasClient2_7, Pathway>execute(atlases27, c -> c.getPathways(PAGE_SIZE).getContent());
+        return pathways.stream()
+                .map(pathway -> conversionService.convert(pathway, CommonPathwayDTO.class))
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<MultipartFile> getAtlasObject(String guid) {
+
+        return commonEntityService.findByGuid(guid).map(entity -> {
+            Atlas origin = entity.getOrigin();
+            JsonNode design = atlasService.<AtlasClient2_7, JsonNode>execute(origin, atlasClient -> atlasClient.exportPathwayDesign(entity.getLocalId()));
+            List<MultipartFile> files = new ArrayList<>();
+            try {
+							String prettyPrint;
+							try {
+								ObjectMapper mapper = new ObjectMapper();
+								prettyPrint = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(design);
+							} catch (JsonProcessingException e) {
+								LOGGER.error("JSON Pretty print failed.");
+								throw new RuntimeJsonMappingException("JSON Pretty print failed.");
+							}
+							// Pathways design
+							MultipartFile designFile = new MockMultipartFile(DESIGN_FILENAME, DESIGN_FILENAME, MediaType.APPLICATION_JSON_VALUE,
+											prettyPrint.getBytes());
+							// Target cohorts
+							List<CohortDefinition> cohortDefinitions = new ArrayList<>();
+							JsonNode targetCohortsNode = design.get("targetCohorts");
+							if (targetCohortsNode instanceof ArrayNode) {
+								cohortDefinitions.addAll(addCohorts(origin, files, (ArrayNode) targetCohortsNode));
+							}
+							// Event cohorts
+							JsonNode eventCohortsNode = design.get("eventCohorts");
+							if (eventCohortsNode instanceof ArrayNode) {
+								cohortDefinitions.addAll(addCohorts(origin, files, (ArrayNode) eventCohortsNode));
+							}
+							files.add(designFile);
```

Comment:
Why do you need cohorts as separate JSONs if you can work off single self-contained Pathways Spec JSON?

Reply:
Oh, here's cohort SQL import. I guess I can use circe to generate cohort SQL and avoid multiple calls to WebAPI, but separate cohort SQLs still remains.